src/main/java/org/spongepowered/api/event/command/CommandProcessEvent.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.command;
+
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.Event;
+
+/**
+ * Fired when a command is processed
+ */
+public interface CommandProcessEvent extends Event {
+
+    /**
+     * Fired before the command is processed
+     */
+    public interface Before extends CommandProcessEvent, Cancellable {}
+
+    /**
+     * Fired after the command is processed
+     */
+    public interface After extends CommandProcessEvent {
+
+        /**
+         * The result of the command.
+         *
+         * @return The result of the command
+         */
+        CommandResult getResult();
```

Comment:
Why? This event is fired after the command is processed, which allows a result to be present; Pre is fired before the command is processed, meaning no result has been produced yet.